### DISCUSSION TITLE
Having strange problems with the new PageData type

### DISCUSSION THREAD

**Discussion by stukennedy** [sveltejs/kit#6077] [Q&A]
when I do the following

```typescript
import type { PageData } from './$types';

export let data: PageData;
$: ({ projects } = data);
```

I get a Typescript error that `Property 'projects' does not exist on type 'PageData'.` , but when I hover over the `PageData` type it can see the variable fine. ( see screenshot)
This is happening on all my pages in VS Code. What am I doing wrong?

<img width="863" alt="Screenshot 2022-08-19 at 13 48 24" src="https://user-images.githubusercontent.com/1593397/185623240-9d14a878-0c6e-4213-a0cb-2897dc8f3739.png">

**Comment by MilesWellsSVT** [comment]
This is a TS issue. Your `PageData` is a union type. For that destructure to work each item in the union would have to contain `projects`. In your loader it is likely you should at least be returning an empty array for `projects` to ensure `projects` always has a value.

ETA: You should also be using [error pages](https://kit.svelte.dev/docs/routing#error) and SK's [loader errors](https://kit.svelte.dev/docs/load#errors) rather than handling your headers and status codes in your page.

**Reply by stukennedy** [reply]
okay, problem is I'm using Supabase Auth Helper, so I'm not in control of the return
```typescript
export async function load({ request, locals: { user } }: RequestEvent) {
	return withApiAuth({ redirectTo: '/login', user }, async () => {
		const projects: Project[] = await apiGetProjects(request);
		return { projects };
	});
}
```
Probably not going to be compatible anymore with these changes.
Doubt I'll need the helpers stuff if I rewrite this.

**Reply by MilesWellsSVT** [reply]
Instead of returning `withApiAuth` you could handle the results yourself in the loader and utilize the status and headers it returns to feed into the SK `error` and `redirect` utilities. You only want to return data from `load` when it's successful.
